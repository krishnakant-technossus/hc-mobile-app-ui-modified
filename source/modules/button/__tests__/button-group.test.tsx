// npx jest ./source/modules/button/__tests__/button-group.test.ts

import * as React from 'react';
import {render} from '@testing-library/react-native';

import Button from '../button';
import ButtonGroup from '../button-group';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('ButtonGroup', () => {
  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <ButtonGroup>
            <Button>Simple Button 1</Button>
            <Button>Simple Button 2</Button>
          </ButtonGroup>
        </ThemeProvider>,
      );

      const buttonGroup = wrapper.getByTestId('button-group');

      expect(buttonGroup).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders its children correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <ButtonGroup>
            <Button>Simple Button 1</Button>
            <Button>Simple Button 2</Button>
          </ButtonGroup>
        </ThemeProvider>,
      );

      const buttons = wrapper.getAllByTestId('button');

      expect(buttons[0]).toBeDefined();
      expect(buttons[1]).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });
  });
});
