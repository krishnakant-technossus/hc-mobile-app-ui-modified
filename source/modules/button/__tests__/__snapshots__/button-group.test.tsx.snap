// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup Rendering renders its children correctly 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "width": "100%",
      },
    ]
  }
  testID="button-group"
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 0,
          "marginRight": 11,
        },
      ]
    }
    testID="button-group-item"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00AEFF",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderWidth": 0,
          "display": "flex",
          "height": 44,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 0,
          "width": "100%",
        }
      }
      testID="button"
    >
      <Text
        appearance="filled"
        color="primary"
        disabled={false}
        small={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "500",
            },
          ]
        }
        testID="button-label"
      >
        Simple Button 1
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 11,
          "marginRight": 0,
        },
      ]
    }
    testID="button-group-item"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00AEFF",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderWidth": 0,
          "display": "flex",
          "height": 44,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 0,
          "width": "100%",
        }
      }
      testID="button"
    >
      <Text
        appearance="filled"
        color="primary"
        disabled={false}
        small={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "500",
            },
          ]
        }
        testID="button-label"
      >
        Simple Button 2
      </Text>
    </View>
  </View>
</View>
`;

exports[`ButtonGroup Rendering renders itself correctly 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "width": "100%",
      },
    ]
  }
  testID="button-group"
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 0,
          "marginRight": 11,
        },
      ]
    }
    testID="button-group-item"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00AEFF",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderWidth": 0,
          "display": "flex",
          "height": 44,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 0,
          "width": "100%",
        }
      }
      testID="button"
    >
      <Text
        appearance="filled"
        color="primary"
        disabled={false}
        small={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "500",
            },
          ]
        }
        testID="button-label"
      >
        Simple Button 1
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginLeft": 11,
          "marginRight": 0,
        },
      ]
    }
    testID="button-group-item"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00AEFF",
          "borderColor": "transparent",
          "borderRadius": 4,
          "borderWidth": 0,
          "display": "flex",
          "height": 44,
          "justifyContent": "center",
          "opacity": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 0,
          "width": "100%",
        }
      }
      testID="button"
    >
      <Text
        appearance="filled"
        color="primary"
        disabled={false}
        small={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "500",
            },
          ]
        }
        testID="button-label"
      >
        Simple Button 2
      </Text>
    </View>
  </View>
</View>
`;
